codegen/compile_protos.sh

```diff
@@ -65,6 +66,17 @@ $PROTOC $PROTOS --python_out "$BUILD_DIR/py_out" $PROTOFLAGS --grpc_out "$BUILD_
 
 $PROTOC $PROTOS --go_out="$BUILD_DIR/go_out" --go-grpc_out="$BUILD_DIR/go_out" $PROTOFLAGS
 
+$PROTOC $PROTOS $PROTOFLAGS \
+    --prost_out="$BUILD_DIR/rust_out/src" \
+    --prost_opt=compile_well_known_types \
+    --prost_opt=extern_path=.google.protobuf=::pbjson_types \
+    --prost-serde_out="$BUILD_DIR/rust_out/src" \
```

Comment:
I am not convinced that you should run this plugin:
- it generates a huge amount of code, most of the code in this PR comes from `.serde.rs` files. Is this plugin responsible for generating the integrality of these files?
- is it really just to convert JSON to an in-memory Protobuf message? Is there no other way to do that without generating a ton of code?

Basically, is there a strong reason to use `protoc-gen-prost-serde` in our case?

Reply:
`serde` is a useful crate for serialize and deserialize. Thus, generating such code enables converting types from and into json, toml, and xml, etc.

I haven't noticed that it generates tons of code. And we may only need such feature for a few types.

I will try to find out if some simple way exist.

Reply:
I tried a simple way mentioned in [`prost-build`](https://docs.rs/prost-build/latest/prost_build/struct.Config.html#method.type_attribute). The generated code can be built, but it **cannot deserialize** `p4info.json` generated by p4c.

I assume that tons of code is needed to generate serialization code that follows the conventions of protobuf. [Ref1](https://github.com/tokio-rs/prost/issues/75#issuecomment-930002425) [Ref2](https://github.com/tokio-rs/prost/issues/75#issuecomment-1383237651)

In conclusion, there are two ways to support JSON:
1. Add `type_attribute` or something in the `protoc-gen-prost` option. This is simple, but it does not follow the conventions of protobuf.
2. Use `protoc-gen-prost-serde`, which is based on `pbjson_build`. It follows the conventions but generates tons of codes.

There may be other ways I have yet to find out.

---

However, from my experience of building a control plane in Rust, we can only use `p4info.bin` and the `decode` method provided by `prost` to serialize P4Info from it.

In addition, JSON support is on `prost`'s [Roadmap](https://github.com/tokio-rs/prost/issues/624).

I think we can remove `serde` (JSON) support for now and wait for the `prost` community to find a better way.

Reply:
Did you try tonic-build ? 

If I understood you correctly, prost-build generated files are not able to deserialize json files (p4info). 

While I did not try it with the p4runtime, I used tonic-build with bfruntime (Tofino) proto definition and was able to deserialize the corresponding bfrt.json perfectly fine.

Reply:
> I used tonic-build with bfruntime (Tofino) proto definition and was able to deserialize the corresponding bfrt.json

Could you provide some sample code for generating code and using it to deserialize? I thought `tonic-build` is based on `prost-build` and doesn't provide deserialization. I wonder what configuration I have missed.

I tried `tonic-build` before the protoc approach in this PR, but I didn't try deserialization then. I only tested `prost::Message::decode`.